store/sqlstore/upgrade.go

```diff
@@ -818,8 +819,73 @@ func upgradeDatabaseToVersion525(sqlStore SqlStore) {
 func upgradeDatabaseToVersion526(sqlStore SqlStore) {
 	// TODO: uncomment when the time arrive to upgrade the DB for 5.26
 	//if shouldPerformUpgrade(sqlStore, VERSION_5_25_0, VERSION_5_26_0) {
+	if err := precheckMigrationToVersion526(sqlStore); err != nil {
```

Comment:
Thinking about this some more, I'm assuming the precheck is to prevent MySQL's default (!) behaviour of silently truncating a field if it's too long.

From reading the docs on [ALTER TABLE](https://dev.mysql.com/doc/refman/5.7/en/alter-table.html), it looks like we could get this checking "for free" with strict mode:
> This conversion may result in alteration of data. For example, if you shorten a string column, values may be truncated. To prevent the operation from succeeding if conversions to the new data type would result in loss of data, enable strict SQL mode before using ALTER TABLE (see Section 5.1.10, “Server SQL Modes”). 

Would it be worth pivoting to that instead of trying to do our own check? I think this also gets us a transactional guarantee, as compared to checking ourselves and risking a (incredibly unlikely) race from a concurrent server writing.

(I'm also assuming Postgres "does the right thing" here.)

Reply:
The problem with that we could end with the DB  partially upgraded which is not something we want. 

Eg We execute the migration and  while the first two alters work the third could fail, you will get an error in Strict mode but at that time you’ve altered two columns, because MySQL does auto commit and doesn’t support DDL transaction.

The idea of this precheck is to test all the affected columns and not begin the migration if some of them is going to fail.

Reply:
@ethervoid, agree that we can't do the schema change transactionally across multiple columns. Admittedly, this has been a problem for every migration we've ever done. Is there something special about this case that warrants more scrutiny? My hope is that this is still a backwards compatible change in that `VERSION-1` can still read/write to these altered column types.

Reply:
> Is there something special about this case that warrants more scrutiny?

This is a multiple alter situation and we should try to do it atomically. Why? as you said if we left this inconsistent for some users shouldn't be a problem because it should be compatible with version-1. But think about what happens if we revert the change, we don't have downgrade script, and in the future, this inconsistency is breaking something.

From my PoV, all the MM users must have the same database schema to avoid problems doing migrations in Vanilla MM.

>  I think this also gets us a transactional guarantee, as compared to checking ourselves and risking an (incredibly unlikely) race from a concurrent server writing

I thought about it and I think a good midterm solution is to include precheck and also use the strict SQL mode

What do you think?

Reply:
> From my PoV, all the MM users must have the same database schema to avoid problems doing migrations in Vanilla MM.

Within minor versions, we actually explicitly support backwards compatibility: it's the reason why `community-daily` / `community-release` and `community` all run against the same database without issue. The only real requirement is that the migration is idempotent, so that if it does fail midway, it can be restarted from the beginning without issue.

> I thought about it and I think a good midterm solution is to include precheck and also use the strict SQL mode.

This seems reasonable! :)

Reply:
Included the new check. I've only add it for this migration but if we don't get any problems with it we can apply to all the migrations